code/game/objects/objs.dm

```diff
@@ -29,7 +29,8 @@
 
 /obj/New()
 	..()
-
+	if (!armor)

```

Comment:
thoughts on an ```armor_params``` params string?, ```"melee=0;bullet=0;laser=0;etc...;"``` (possibly validated by travis, for typos), and the actual ```armor``` list being built from these params strings on first use (some kind of ```get_armour()``` proc, that defaults to returning a static ```list(melee = 0, bullet = 0, laser = 0, energy = 0, bomb = 0, bio = 0, rad = 0, fire = 0, acid = 0)``` for items without the ```armor_params```